kube/secret.go

```diff
@@ -14,42 +14,39 @@ import (
 // showing which original CV name maps to what secret+key combination.
 func MakeSecrets(secrets model.CVMap, settings ExportSettings) (helm.Node, error) {
 	data := helm.NewMapping()
-	for name, cv := range secrets {
-		var value string
+	generated := helm.NewMapping()
 
+	for name, cv := range secrets {
 		key := util.ConvertNameToKey(name)
+		var value interface{}
+		comment := cv.Description
 
 		if settings.CreateHelmChart {
-			if settings.UseSecretsGenerator && cv.Generator != nil {
-				// it will be created during runtime
-				continue
-			}
-			if cv.Generator != nil && cv.Generator.Type == model.GeneratorTypePassword {
-				value = "{{ randAlphaNum 32 | b64enc | quote }}"
-			} else {
-				errString := fmt.Sprintf("%s configuration missing", cv.Name)
-				value = fmt.Sprintf(`{{ required "%s" .Values.env.%s | b64enc | quote }}`, errString, cv.Name)
+			if cv.Generator == nil {
+				errString := fmt.Sprintf("secrets.%s has not been set", cv.Name)
+				value = fmt.Sprintf(`{{ required "%s" .Values.secrets.%s | b64enc | quote }}`, errString, cv.Name)
+				if cv.Immutable {
+					comment += "\nThis value is immutable and must not be changed once set."
+				}
+				data.Add(key, helm.NewNode(value, helm.Comment(comment)))
+			} else if !cv.Immutable {
```

Comment:
What happens if there is a generator and it is immutable?

Reply:
It was a conscious decision (by me & @viovanov) to make immutable secrets with a generator not overridable by the user.  There should never be a need to do so. Therefore these secrets don't need to be included in `values.yaml` or `templates/secrets.yaml`.

Personally I think *none* of the generated secrets need to be user-overridable if we have a working rotation mechanism, but @viovanov did not agree, so the rest are still there...

Reply:
Then please add a comment (or better, an assertion) to show that we thought about the case (and rejected it).